### ISSUE TITLE
Universal and Tex examples broken for 0.9.1 entity syntax

### ISSUE THREAD

**Issue by flarnie** [facebook/draft-js#1668]
**Do you want to request a *feature* or report a *bug*?**
This is a bug, either with our example files or with Draft.js itself.

**What is the current behavior?**
On master if you run the 'universal' example under `examples/draft-0-9-1/` and try typing into the editor, the editor does not update and errors are thrown in the console:
![screen shot 2018-02-26 at 8 21 26 am](https://user-images.githubusercontent.com/1114467/36681793-729e7984-1ace-11e8-8294-7d4ad8da23d5.png)

Similarly if you run the 'tex' example under `examples/draft-0-9-1/` and try typing into the editor, various errors are thrown and things can quickly get into an invalid state, where wrong characters and selection state are created:
![screen shot 2018-02-23 at 4 15 53 pm](https://user-images.githubusercontent.com/1114467/36681888-a91718f4-1ace-11e8-8542-286c1d035b90.png)

**If the current behavior is a bug, please provide the steps to reproduce and if possible a minimal demo of the problem. You can use this jsfiddle to get started: https://jsfiddle.net/stopachka/m6z0xn4r/.**

`cd examples/draft-0-9-1/tex && yarn install && yarn start` then visit localhost:3000 and try editing the 'tex' examples and the text.

`cd examples/draft-0-9-1/universal && yarn install && yarn build && yarn start` then visit localhost:3003 and try typing text.

**NOTE:** The problem is only present in the 0.9.1 examples, not the 0.10.1 examples.

**What is the expected behavior?**

Entering text should update the editor as normal, not throw errors.

**Which versions of Draft.js, and which browser / OS are affected by this issue? Did this work in previous versions of Draft.js?**
Tested with the master branch in Firefox.

**Comment by flarnie** [comment]
I'd like to verify that this isn't a bug in Draft.js before doing the next release.

**Comment by bambii7** [comment]
This is happening because Tex & Universal are the only examples with their own package.json. Referencing Rect & React-Dom  `^15.4.1` while referencing the parent projects Draftjs which uses `^16.0.0` versions. The mismatch causes modules to not mount correctly and all the state issues.

```
"draft-js": "file:../../../",
...
"react": "^15.4.1",
 "react-dom": "^15.4.1"
```

I believe referencing the parent projects react & react-dom would be the way to go. And some example tests. Otherwise whenever draft updates its react versions this issue will occur again. 

```
"react": "file:../../../node_modules/react",
"react-dom": "file:../../../node_modules/react-dom"
```

**Comment by niveditc** [comment]
Fixed in #1691
